### ISSUE TITLE
_release.filePathDelta is broken

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4282]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#4282|
|Type|defect|
|Created|2010-02-19 18:33:31Z|
|Branch|https://github.com/twisted/twisted/tree/fix-filepath-delta-4282|

For example, doc/core/howto and doc/web/howto report 2 common path elements, instead of only one (because doc==doc, and howto==howto).

<details><summary>Searchable metadata</summary>

```
trac-id__4282 4282
type__defect defect
reporter__therve therve
priority__normal normal
milestone__ 
branch__branches_fix_filepath_delta_4282 branches/fix-filepath-delta-4282
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__ 
time__1266604411000000 1266604411000000
changetime__1266763787000000 1266763787000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

(In [28283]) Branching to 'fix-filepath-delta-4282'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="free.ekanayaka's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|free.ekanayaka set owner to @therve|
|-|-|

Looks great, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set status to `closed`|
|-|-|

(In [28291]) Merge fix-filepath-delta-4282

Author: therve
Reviewer: free.ekanayaka
Fixes: [#4282](https://github.com/twisted/twisted/issues/4282)

Fix a bug in _release.filePathDelta when paths are sharing common ends but
differ in the middle.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
